plugin/src/main/java/org/opensearch/ml/action/stats/MLStatsNodesTransportAction.java

```diff
@@ -127,21 +156,89 @@ MLStatsNodeResponse createMLStatsNodeResponse(MLStatsNodesRequest mlStatsNodesRe
         }
 
         Map<String, MLModelStats> modelStats = new HashMap<>();
-        // return model level stats
         if (mlStatsInput.includeModelStats()) {
-            for (String modelId : mlStats.getAllModels()) {
-                if (mlStatsInput.retrieveStatsForModel(modelId)) {
-                    Map<ActionName, MLActionStats> actionStatsMap = new HashMap<>();
-                    for (Map.Entry<ActionName, MLActionStats> entry : mlStats.getModelStats(modelId).entrySet()) {
-                        if (mlStatsInput.retrieveStatsForAction(entry.getKey())) {
-                            actionStatsMap.put(entry.getKey(), entry.getValue());
+            CountDownLatch latch = new CountDownLatch(1);
+            boolean isSuperAdmin = isSuperAdminUserWrapper(clusterService, client);
+            searchHiddenModels(ActionListener.wrap(hiddenModels -> {
```

Comment:
If isSuperAdmin is true, de we need to search hidden models?

Reply:
I was thinking to add a flag of the hidden model based on that search so that if superadmin pulls the metrics they know which one is the hidden model metrics, but I was thinking to add that flag in another PR, not in this PR.

Reply:
Got it, I think you mean add is_hidden flag in stats API response, so user can know the hidden model usage